src/utils/path.ts

```diff
@@ -110,3 +111,24 @@ export const getWorkspacePath = (defaultCwdPath = "") => {
 	}
 	return cwdPath
 }
+
+export const isLocatedInWorkspace = (pathToCheck: string = ""): boolean => {
+	const workspacePath = getWorkspacePath()
+
+	// Handle long paths in Windows
+	if (pathToCheck.startsWith("\\\\?\\") || workspacePath.startsWith("\\\\?\\")) {
+		return pathToCheck.startsWith(workspacePath)
+	}
+
+	const resolvedPath = path.resolve(workspacePath, pathToCheck)
+
+	// Using realpathSync to resolve any symbolic links
```

Comment:
You shouldn't resolve symbolic links like this can lead to a directory traversal attack